### PR TITLE
CODE RUB: Remove OpenAPI hard coding

### PR DESCRIPTION
**Pull request by BrianLParker**
This PR moves all the OpenAPI (Swagger) document configuration to `appsettings.Development.json`. 

swagger

### CODE REVIEW

File: Taarafo.Core/Startup.cs [hunk 1]
```diff
@@ -25,38 +29,36 @@ public class Startup
 
         public void ConfigureServices(IServiceCollection services)
         {
+            LocalConfigurations localConfigurations = Configuration.Get<LocalConfigurations>();
+            services.AddScoped<LocalConfigurations>(sp => localConfigurations);
+
```
Comment (ShriHumrudha) [inline Taarafo.Core/Startup.cs:34]: Line 34 can be removed

File: Taarafo.Core/Startup.cs [hunk 2]
```diff
@@ -25,38 +29,36 @@ public class Startup
 
         public void ConfigureServices(IServiceCollection services)
         {
+            LocalConfigurations localConfigurations = Configuration.Get<LocalConfigurations>();
+            services.AddScoped<LocalConfigurations>(sp => localConfigurations);
```
Comment (ShriHumrudha) [inline Taarafo.Core/Startup.cs:33]: what is sp here? Can we have a more readable name?

File: Taarafo.Core/Startup.cs [hunk 3]
```diff
@@ -25,38 +29,36 @@ public class Startup
...
 
                 app.UseSwaggerUI(options =>
                 {
-                    options.SwaggerEndpoint(
-                        url: "/swagger/v1/swagger.json",
-                        name: "Taarafo.Core v1");
+                    foreach (var openAPIConfiguration in openAPIConfigurations)
+                    {
+                        options.SwaggerEndpoint(
+                            url: openAPIConfiguration.OpenAPIEndpoint.Url,
```
Comment (ShriHumrudha) [inline Taarafo.Core/Startup.cs:59]: We can remove the parameter aliasing as the ending property name is same as the parameter name.

File: Taarafo.Core/Startup.cs [hunk 4]
```diff
@@ -75,5 +77,43 @@ private static void AddBrokers(IServiceCollection services)
...
+            IServiceCollection services,
+            LocalConfigurations localConfigurations)
+        {
+            IEnumerable<OpenAPIConfiguration> openAPIConfigurations = localConfigurations.OpenAPIs;
+
+            foreach (OpenAPIConfiguration openAPIConfiguration in openAPIConfigurations)
+            {
+                string version = openAPIConfiguration.Version;
+                OpenAPIDocumentConfiguration document = openAPIConfiguration.Document;
+                services.AddSwaggerGen(options =>
```
Comment (ShriHumrudha) [inline Taarafo.Core/Startup.cs:91]: Need a line break here and inside the OpenApiInfo model too.

File: Taarafo.Core/Startup.cs [hunk 5]
```diff
@@ -75,5 +77,43 @@ private static void AddBrokers(IServiceCollection services)
...
+                {
+                    OpenApiInfo openApiInfo = new OpenApiInfo
```
Comment (ShriHumrudha) [inline Taarafo.Core/Startup.cs:93]: left hand side can be var as the right hand side is clear.

File: Taarafo.Core/Startup.cs [hunk 6]
```diff
@@ -75,5 +77,43 @@ private static void AddBrokers(IServiceCollection services)
...
+                        },
+                        License = new OpenApiLicense
+                        {
+                            Name = document.LicenseName,
+                            Url = new Uri(document.LicenseUrl),
+                        }
+                    };
+
+                    options.SwaggerDoc(name: version, info: openApiInfo);
+
```
Comment (ShriHumrudha) [inline Taarafo.Core/Startup.cs:113]: This line can be removed.

**Review by ShriHumrudha — CHANGES_REQUESTED** [review]

**Comment by hassanhabib** [comment]
Closing due to no activity

**Pull request closed without merging**